compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -8109,7 +8109,7 @@ private BType checkOptionalFieldAccessExpr(BLangFieldBasedAccess fieldAccessExpr
         BType actualType = symTable.semanticError;
 
         boolean nillableExprType = false;
-        BType effectiveType = varRefType;
+        BType effectiveType = Types.getReferredType(varRefType);
```

Comment:
We are doing this check when creating `varRefType`. https://github.com/ballerina-platform/ballerina-lang/blob/79771f7ffacc44a984f6b165ff0557709c6d819f/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java#L2101
So we can change L2109(https://github.com/ballerina-platform/ballerina-lang/blob/79771f7ffacc44a984f6b165ff0557709c6d819f/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java#L2109) as
`return Objects.requireNonNullElse(effectiveType, type);`

Reply:
We want `varRefType` as it is now.
Because we use it to display error messages for user.

Reply:
I will add

Reply:
Added 58a50b349bb9a75fb4b4c16241fdefb940bca435